cmake/app/boilerplate.cmake

```diff
@@ -247,7 +247,15 @@ else() # NOT FIRST_BOILERPLATE_EXECUTION
 
   # Have the child image select the same BOARD that was selected by
   # the parent.
+  # Unless parent was "ns" in which case we assume that the child images are
+  # all are secure, and should be build using the secure version of the board.
```

Comment:
We need a TODO here explaining what we are assuming, why this is an invalid assumption, and how we intend to re-write this in the future.

Reply:
will add

Reply:
check it now